### PR TITLE
msg expiry feature

### PR DESCRIPTION
**Pull request by JaylinYu**

### CODE REVIEW

File: nanomq/apps/broker.c [hunk 1]
```diff
@@ -431,41 +430,34 @@ server_cb(void *arg)
...
-				nng_msg_clone(smsg);
-				work->pid.id = msg_info->pipe;
-				nng_msg_set_pipe(smsg, work->pid);
-				work->msg = NANO_NNI_LMQ_PACKED_MSG_QOS(smsg, msg_info->qos);
-				nng_aio_set_prov_data(work->aio, (void *)msg_info->retain);
-				nng_aio_set_msg(work->aio, work->msg);
-				nng_ctx_send(work->ctx, work->aio);
-				nng_aio_set_prov_data(work->aio, NULL);
-			}
+				if (encode_pub_message(smsg, work, PUBLISH, msg_info))
```
Comment (wanghaEMQ) [inline nanomq/apps/broker.c:434]: Actually, encode_pub_message always returns true.

File: nanomq/apps/broker.c [hunk 2]
```diff
@@ -431,41 +430,34 @@ server_cb(void *arg)
...
+						nng_msg_clone(smsg);
+						work->pid.id = msg_info->pipe;
+						nng_msg_set_pipe(smsg, work->pid);
+						work->msg = NANO_NNI_LMQ_PACKED_MSG_QOS(smsg, msg_info->qos);
+						nng_aio_set_prov_data(work->aio, (void *)msg_info->retain);
+						nng_aio_set_msg(work->aio, work->msg);
+						nng_ctx_send(work->ctx, work->aio);
+						nng_aio_set_prov_data(work->aio, NULL);
+					}
 			nng_msg_free(smsg);
```
Comment (wanghaEMQ) [inline nanomq/apps/broker.c:446]: we assume encode function would return false in the future, this message may be null if encode failed.

**Review by wanghaEMQ — APPROVED** [review]
Have u formatted your code?

File: nanomq/apps/broker.c (hunk 2 above)
Comment (JaylinYu) [reply nanomq/apps/broker.c:446]: ok will do

File: nanomq/apps/broker.c (hunk 1 above)
Comment (JaylinYu) [reply nanomq/apps/broker.c:434]: not in this PR

**Review by wanghaEMQ — COMMENTED** [review]
good

**Pull request merged**